pkg/keccak/patches/0001-squash.patch

```diff
@@ -0,0 +1,108 @@
+ * SP800-185: all the functions defined in the NIST SP 800-185 standard, i.e., cSHAKE, TupleHash, KMAC and ParallelHash.
+ * KeccakDuplexWidth{200, 400, 800, 1600}: the Keccak[r, c] duplex objects, where r+c equals the given permutation width (200, 400, 800 or 1600).
+@@ -60,6 +62,17 @@ The fragments below allow to select the desired cryptographic functions (a.k.a.
+     </fragment>
+ 
+     <fragment name="KeccakSponge" inherits="KeccakSpongeWidth200 KeccakSpongeWidth400 KeccakSpongeWidth800 KeccakSpongeWidth1600"/>
++
++	<!-- Custom -->
++	<fragment name="Keccak800" inherits="KeccakSpongeWidth800">
++		<c>Modes/keccak800.c</c>
++		<h>Modes/keccak800.h</h>
++	</fragment>
++
++	<fragment name="Keccak1600" inherits="KeccakSpongeWidth1600">
++		<c>Modes/KeccakHash.c</c>
++		<h>Modes/keccak1600.h</h>
++	</fragment>
+ 
+     <!-- NIST standards -->
+ 
+diff --git a/LowLevel.build b/LowLevel.build
+index 3ea6b72..b26cf4b 100644
+--- a/LowLevel.build
++++ b/LowLevel.build
+@@ -114,15 +114,11 @@ The fragments below allow to select the desired implementation of the permutatio
+ 
+     <!-- *** Low-level services *** -->
+ 
+     <fragment name="optimized">
+-        <h>Common/brg_endian.h</h>
+-        <gcc>-fomit-frame-pointer</gcc>
+         <gcc>-O2</gcc>
++        <gcc>$(CFLAGS)</gcc>
+-        <gcc>-g0</gcc>
+-        <gcc>-march=native</gcc>
+-        <gcc>-mtune=native</gcc>
+     </fragment>
+ 
+     <!-- Keccak-p[200] -->
+ 
+     <fragment name="width200">
+diff --git a/Makefile.build b/Makefile.build
+index fe6b6dc..f65e423 100644
+--- a/Makefile.build
++++ b/Makefile.build
+@@ -166,4 +167,9 @@ http://creativecommons.org/publicdomain/zero/1.0/
+         </product>
+     </group>
+ 
++    <!-- Hashing with Keccak-800 or Keccak-1600 -->
++    <target name="KeccakHashARMv6M.a" inherits="Keccak800 optimized800ARMv6Mu1 Keccak1600 optimized1600ARMv6Mu1"/>
++    <target name="KeccakHashARMv7M.a" inherits="Keccak800 optimized800ARMv7Mu2 Keccak1600 inplace1600ARMv7M"/>
++    <target name="KeccakHashNative.a" inherits="Keccak800 optimized800ufull Keccak1600 compact1600"/>
++
+ </build>
+diff --git a/Modes/KeccakHash.h b/Modes/KeccakHash.h
+index 0d12975..809e929 100644
+--- a/Modes/KeccakHash.h
++++ b/Modes/KeccakHash.h
+@@ -19,16 +19,12 @@ http://creativecommons.org/publicdomain/zero/1.0/
+ #ifndef KeccakP1600_excluded
+ 
+ #include "KeccakSpongeWidth1600.h"
++#include "keccak_defs.h"
+ #include <string.h>
+ 
+-#ifndef _Keccak_BitTypes_
+-#define _Keccak_BitTypes_
+-typedef unsigned char BitSequence;
+-
+-typedef size_t BitLength;
+-#endif
+-
+-typedef enum { SUCCESS = 0, FAIL = 1, BAD_HASHLEN = 2 } HashReturn;
++#define BitSequence bit_sequence
++#define BitLength   bit_length
++#define HashReturn  hash_return
```

Comment:
Is this patch really necessary?

Reply:
Yes and no - if we do without the macros for BitSequence and BitLength, we would need to define those variables in keccak800.h with those names instead of bit_sequence and bit_length, which I'd rather not because the original names don't respect RIOT's naming conventions. Plus, HashReturn has been moved to a separate header so that we can use it for both keccak800.h and keccak1600.h (instead of needing to define a separate HashReturn for keccak800.h, which is wholly silly), which makes a patch removing the definition of HashReturn inside KeccakHash.h necessary. I feel that if we need to do a patch to that file anyway, we could also insert those macros to allow for a consistent naming. This allows us to use the names according to RIOT's naming convention inside our new header files (which are part of the RIOT package) while not needing to rename them inside the original code from KeccakCodePackage (which is pulled and not part of RIOT). So it all works out!

Reply:
I need to dig a little into the code, but I think it would be a lot simpler and straight-forward to include `Modes/KeccakHash.h` into `keccak_defs.h`.

Reply:
Now that Iooked into `Modes/KeccakHash.h`, I still think, that what I proposed yesterday makes more sense. So please include this file into `keccak_defs.h`, and do the `#define`s there.

Reply:
After further consideration, I think I agree with you.